packages/amplify_datastore_plugin_interface/lib/src/types/models/auth_rule.dart

```diff
@@ -20,23 +20,22 @@ import 'package:collection/collection.dart';
 import 'model.dart';
 
 class AuthRule {
-  // enum
   final AuthStrategy authStrategy;
-  final String ownerField;
-  final String identityClaim;
-  final String groupClaim;
+  final String ownerField; //opt
+  final String identityClaim; //opt
+  final String groupClaim; //opt
   final List<String> groups;
-  final String groupsField;
-  final List<ModelOperation> operations;
+  final String groupsField; //opt
+  final List<ModelOperation> operations; //opt
 
   const AuthRule(
       {this.authStrategy,
-      this.ownerField,
-      this.identityClaim,
-      this.groupClaim,
-      this.groups,
-      this.groupsField,
-      this.operations});
+      this.ownerField = "",
+      this.identityClaim = "",
+      this.groupClaim = "",
+      this.groups = const [],
+      this.groupsField = "",

```

Comment:
Same here, I think null is probably a better default for the strings.

Reply:
See comment above.  null will break platform serialization

Reply:
Ah yes, makes sense.